src/test/java/seedu/multitasky/logic/CommandHistoryTest.java

```diff
@@ -16,12 +24,50 @@ public void setUp() {
     }
 
     @Test
-    public void add() {
+    public void init_success() {
+        Calendar expectedStartDate = null;
+        Calendar expectedEndDate = null;
+        Set<String> expectedKeywords = new HashSet<String>();
+        Entry.State expectedState = Entry.State.ACTIVE;
+        assertAll(history, expectedKeywords, expectedStartDate, expectedEndDate, expectedState);
+    }
+
+    @Test
+    public void add_success() {
         final String validCommand = "clear";
         final String invalidCommand = "adds Task";
 
         history.add(validCommand);
         history.add(invalidCommand);
         assertEquals(Arrays.asList(validCommand, invalidCommand), history.getHistory());
     }
+
+    @Test
+    public void setPrevSearch_validArgs_success() {
+        Calendar startDate = Calendar.getInstance();
+        Calendar endDate = Calendar.getInstance();
+        Set<String> keywords = new HashSet<String>(Arrays.asList("new keywords"));
+        Entry.State state = Entry.State.ARCHIVED;
+        history.setPrevSearch(keywords, startDate, endDate, state);
+        assertAll(history, keywords, startDate, endDate, state);
+    }
+
```

Comment:
I think more like the private helper method should be all the way at the bottom rather than in the middle of the public test methods